internal/git/commands.go

```diff
@@ -519,11 +520,26 @@ func (self *Commands) HasLocalBranch(runner gitdomain.Runner, name gitdomain.Loc
 	return runner.Run("git", "show-ref", "--quiet", "refs/heads/"+name.String()) == nil
 }
 
-func (self *Commands) HasMergeInProgress(runner gitdomain.Runner) bool {
-	err := runner.Run("git", "rev-parse", "-q", "--verify", "MERGE_HEAD")
+func (self *Commands) HasMergeInProgress(querier gitdomain.Querier) bool {
+	_, err := querier.Query("git", "rev-parse", "-q", "--verify", "MERGE_HEAD")
 	return err == nil
 }
 
+func (self *Commands) HasRebaseInProgress(querier gitdomain.Querier) bool {
+	gitDir, err := querier.QueryTrim("git", "rev-parse", "--absolute-git-dir")
+	if err != nil {
+		return false
+	}
+	for _, rebaseDirName := range []string{"rebase-merge", "rebase-apply"} {
```

Comment:
This is cool that it detects a rebase by looking at the files that the user needs to edit when doing a rebase. Let's hope that these file names don't change in future Git versions.

Reply:
It's weirdly difficult to tell if a rebase is in progress! I tried looking for `REBASE_HEAD` at first, but it still exists after a rebase is finished. Checking for the presence of the directories `.git/rebase-merge` or `.git/rebase-apply` is how https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh detects if a rebase is in progress.

I think it's really unlikely that this will change in the future because it would break other clients who use this method to detect a rebase. I ran into this PR from VS Code where they also check these folders: https://github.com/microsoft/vscode/pull/57651/files